core/src/main/java/org/elasticsearch/rest/action/RestActions.java

```diff
@@ -243,6 +243,12 @@ public RestResponse buildResponse(NodesResponse response, XContentBuilder builde
     private static QueryBuilder parseTopLevelQueryBuilder(XContentParser parser) {
         try {
             QueryBuilder queryBuilder = null;
+            XContentParser.Token first = parser.nextToken();
+            if (first == null) {
+                return null;
+            } else if (first != XContentParser.Token.START_OBJECT) {
+                throw new ParsingException(parser.getTokenLocation(), "Failed to parse");
```

Comment:
Can you make the message a little bit more descriptive, since we know whats wrong in this case?

Reply:
@cbuescher sure, how about `"Invalid JSON object, first token must be '{' but was 'VALUE_STRING'"` ?

Reply:
It doesn't have to be JSON, we support Yaml, Cbor and Smile, so better to say we expected the beginning of an object or something along those lines.

Reply:
Maybe do sth. similar as this: https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java#L962

Reply:
yea that looks perfect, sec adjusting :)